pkg/build/builder/source.go

```diff
@@ -77,7 +77,9 @@ func fetchSource(dir string, build *api.Build, urlTimeout time.Duration, in io.R
 func checkRemoteGit(url string, timeout time.Duration) error {
 	glog.V(4).Infof("git ls-remote --heads %q", url)
 	cmd := exec.Command("git", "ls-remote", "--heads", url)
-	cmd.Env = []string{"GIT_ASKPASS=/bin/true", "GIT_SSH=" + os.Getenv("GIT_SSH")}
+	env := os.Environ()
+	env = append(env, "GIT_ASKPASS=/bin/true")
```

Comment:
@gabemontero the child process will only get vars you set in the Command.Env field. This is why basic authentication was broken with your change. If you leave the Env field nil, then it will use whatever the parent process has in its environment.